In [1]:
import pandas as pd
import numpy as np

df_bikes = pd.read_csv('../input/data-wrangling-bike-rentals/bike_rentals_cleaned.csv')

df_bikes.head()

instant  season   yr  mnth  holiday  weekday  workingday  weathersit  \
0        1     1.0  0.0     1      0.0      6.0         0.0           2   
1        2     1.0  0.0     1      0.0      0.0         0.0           2   
2        3     1.0  0.0     1      0.0      1.0         1.0           1   
3        4     1.0  0.0     1      0.0      2.0         1.0           1   
4        5     1.0  0.0     1      0.0      3.0         1.0           1   

       temp     atemp       hum  windspeed   cnt  
0  0.344167  0.363625  0.805833   0.160446   985  
1  0.363478  0.353739  0.696087   0.248539   801  
2  0.196364  0.189405  0.437273   0.248309  1349  
3  0.200000  0.212122  0.590435   0.160296  1562  
4  0.226957  0.229270  0.436957   0.186900  1600

In [2]:
X = df_bikes.iloc[:,:-1]
y = df_bikes.iloc[:,-1]

In [3]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=2)

In [10]:
from sklearn.model_selection import cross_val_score
from sklearn.tree import DecisionTreeRegressor

model = DecisionTreeRegressor(random_state=2)
scores = cross_val_score(model, X, y, scoring='neg_mean_squared_error',cv=5)
rmse = np.sqrt(-scores)
print('RMSE:',np.round(rmse,2))
print('RMSE mean: %0.2f' % (rmse.mean()))

RMSE: [1252.27  737.69 1577.64 1142.05 1457.13]
RMSE mean: 1233.36


In [15]:
from sklearn.metrics import mean_squared_error

dec_reg = DecisionTreeRegressor()

dec_reg.fit(X_train,y_train)

y_preds = dec_reg.predict(X_train)

mse = mean_squared_error(y_preds,y_train)

mse = np.sqrt(mse)

mse



0.0

A RMSE of means that the model has perfectly fit every data point! This perfect score combined with a cross-validation error 1233.36 is proof that the decision tree is overfitting the data with high variance. The training set fit perfectly, but the test set missed badly.
Hyperparameters may rectify the situation.